tests/listsAndDropDowns.spec.ts

```diff
@@ -0,0 +1,63 @@
+import { test, expect } from '@playwright/test';
+
+
+test.beforeEach(async({page}) => {
+    await page.goto('https://petclinic.bondaracademy.com/welcome')
+})
+
+test.describe('Lists and Dropdowns', () => {
+    test.beforeEach(async ({page}) => {
+        await page.getByRole('button', {name: ' Owners'}).click() //1. Select owners in nav bar and then click search
+        await page.getByRole('link', {name: ' Search'}).click()
+    })
+test('Validate selected pet types from list', async ({page}) => {
+    const petTypeMenuDropDown = page.locator('#type')
+    await page.goto('https://petclinic.bondaracademy.com/owners')
+    await expect(page.locator('app-owner-list h2')).toHaveText('Owners') //2. Assert Owners text is displayed
+    await page.getByRole('row', {name: "George Franklin"}).getByRole('link', {name: "George Franklin"}).click() //3. Select first owner George Franklin
+    await expect(page.locator('.ownerFullName')).toHaveText('George Franklin') //4. Assert that owner Name value displayed is George Franklin
+    await page.locator('app-pet-list').getByRole('button', {name: 'Edit Pet'}).click() //5. Click Edit button for Leo
+    await expect(page.locator('h2')).toHaveText(' Pet ') //6. Assert Pet is displayed as header
+    await expect(page.locator('#owner_name')).toHaveValue('George Franklin') //7. Assert that George Franklin is displsyed in owner field
+    await expect(page.getByLabel('Type')).toHaveValue('cat') //8. Assert value cat is displayed in type field
+    
+    const pets = {
+        cat: 'cat',
+        dog: 'dog',
+        lizard: 'lizard',
+        snake: 'snake',
+        bird: 'bird',
+        hamster: 'hamster'
+    }
+    await petTypeMenuDropDown.click()
+    for(const pet in pets){ //9. Loop through values in dropdown and assert that every value is displayed in type field
+        await petTypeMenuDropDown.filter({hasText: pet}).selectOption(pet)
+        await expect(petTypeMenuDropDown).toHaveValue(pet)
+        if(pet != 'hamster')
+            await petTypeMenuDropDown.click()
+    }
+})
+
+test('Validate the pet type update', async ({page}) => {
+    await page.goto('https://petclinic.bondaracademy.com/owners')
+    const petTypeMenuDropDown = page.locator('#type')
+    const petNameInputField = page.getByLabel('name')
+    await expect(page.locator('app-owner-list h2')).toHaveText('Owners') //2. Assert Owners text is displayed
+    await page.getByRole('row', {name: "Eduardo Rodriquez"}).getByRole('link', {name: "Eduardo Rodriquez"}).click() //3. Select owner Eduardo Rodriquez
+    await page.getByRole('row', {name: "Rosy"}).getByRole('button', {name: 'Edit Pet'}).nth(1).click() //4. Click Edit button for Rosy
```

Comment:
This locator is kind of incorrect, because it does not really returns the Edit button for the pet Rossy. It return buttons for both pets, but you hardcoding the index of the second button to select the second Edit Pet.
![image](https://github.com/MatthewSingler/playwright-homework/assets/155187576/accb4427-b430-45c1-a9c8-8229968c9196)
With this approach you could so something like `page.getByRole('button', {name: 'Edit Pet'}).nth(1).click()` and it would work the same

The more correct locator in this example would be `await page.locator(".dl-horizontal", {hasText: "Rosy"}).getByRole('button', {name: 'Edit Pet'}).click()`
It really clicks on the Edit Pet button for the Rossy. And if one day Jewel pet will be deleted from this owner, test will still work because locator is not dependent on the index of the button

Reply:
This makes sense. I will work to make locators more specific. Updated